### ISSUE TITLE
HiDPI Support

### ISSUE THREAD

**Issue by dimkr** [vlastavesely/raleigh-reloaded#16]
Hi @vlastavesely!

This wonderful theme is the base the Puppy Linux GTK+ 3 themes (https://github.com/puppylinux-woof-CE?q=gtk3_) inherit from and I'm currently working on HiDPI support in Puppy (https://github.com/puppylinux-woof-CE/woof-CE/issues/3335, among others).

This theme contains many GTK+ 2 era, low-resolution assets that don't work well with scale factor 2. For example, the checkboxes:

![185744196-b95fbcdc-6104-405a-bdac-c487574d7d62](https://user-images.githubusercontent.com/1471149/187060073-c9401e3a-0174-41bc-bae6-eab7f915b428.png)

To make this theme future-proof, they need to be replaced with CSS borders+box shadows (for example, the menu and toolbar borders, or the checkbox background+border) or SVG assets (for example, the V inside the checkbox). I'm working on this and hope to achieve 1:1 visual consistency.

**Comment by vlastavesely** [comment]
Hello,
if I remember correctly, I tried checkboxes based on CSS borders but there were some problems and so I used PNG images. I will check the changes in your code and try to either use something or at least remember where the problem was.

**Comment by dimkr** [comment]
Made some good progress, most of the theme is pixel-perfect with scale factor 2. I doubled the size of all images, so they're rendered at size 100% with scaling factor 2, or size 50% with scaling factor 1.

I still see some blurry border images in notebook tabs :(

![tab1](https://user-images.githubusercontent.com/1471149/189472684-0fb55164-3d2c-4214-b15e-8b285175d7c4.png)
![tab3](https://user-images.githubusercontent.com/1471149/189472685-9f096f9f-5449-46f6-b683-ba1ebe1989df.png)

**Comment by vlastavesely** [comment]
Which desktop environment do you use? MATE, which I use, does not seem to have the scaling option.

**Comment by dimkr** [comment]
> Which desktop environment do you use? MATE, which I use, does not seem to have the scaling option.

I don't use a desktop environment. I use dwl (a Wayland compositor) and set the scaling factor to 2 using `wlr-randr --output whatever --scale 2`.

**Comment by vlastavesely** [comment]
I played with it a bit. It is an interesting solution to double the size of the images. However this will not solve anything when the scaling is different from 200%. Also, I am not fully comfortable with it as it is a very specific thing that I have not seen in any other theme. I tried to make a SVG version but with little-to-no success. I am going to leave this issue opened and we will see whether to use your solution or leave it as it is.

I am not ruling out the possibility of me playing with it a bit in the following months but it is not going to be a priority for me :)
